envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -40,10 +41,20 @@ class EnvoySnapshotFactory(
     ): GlobalSnapshot {
         val sample = Timer.start(meterRegistry)
 
-        val clusters = clustersFactory.getClustersForServices(clusterConfigurations.values, communicationMode)
+        var v2Clusters = emptyList<Cluster>()
+        var v2SecuredClusters = emptyList<Cluster>()
+        if (properties.supportV2Configuration) {
+            v2Clusters = clustersFactory.getClustersForServices(
+                clusterConfigurations.values, communicationMode,
+                ResourceVersion.V2
+            )
+            v2SecuredClusters = clustersFactory.getSecuredClusters(v2Clusters)
```

Comment:
Currently we create v3 clusters by mapping v2 clusters and changing only single field. With this PR we change the behaviour - we generate v2 clusters "from scratch". This may be a quite big change that we didn't test.

Maybe we should stick to tested behaviour and generate v2 cluster by mapping v3 clusters (`clustersFactory.mapToV2Cluster(clusters)`)?

Reply:
done

Reply:
Fixed